sdk/notificationhubs/notification-hubs/src/auth/sasTokenProvider.ts

```diff
@@ -1,16 +1,38 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import {
-  AccessToken,
-  NamedKeyCredential,
-  SASCredential,
-  isNamedKeyCredential,
-  isSASCredential,
-} from "@azure/core-auth";
-import { isObjectWithProperties } from "@azure/core-util";
 import { signString } from "./hmacSha256.js";
 
+/**
+ * Represents a named key credential.
+ */
+export interface NamedKeyCredential {
```

Comment:
curious about the choice of duplicating here v.s. importing from core-auth.

Reply:
Removed core-auth as a dependency for now as it is not needed because of my own interfaces